zds/tutorialv2/views/contents.py

```diff
@@ -1286,7 +1286,10 @@ def form_valid(self, form):
         self.object.update(sha_draft=sha, update_date=datetime.now())
 
         self.success_url = extract.get_absolute_url()
-
+        if self.request.is_ajax():
+            return HttpResponse(json.dumps({'result': 'ok', 'last_hash': extract.compute_hash(),
+                                            'new_url': extract.get_edit_url()}),
+                                content_type='application/json')
```

Comment:
On pourrait utiliser une `JsonResponse`, qui gère automatiquement les détails techniques (sérialisation, `Content-Type`, etc.) (il faudrait ajouter `from django.http import JsonResponse`). C'est aussi plus lisible, je trouve.

```suggestion
            return JsonResponse({'result': 'ok', 'last_hash': extract.compute_hash(),
                                 'new_url': extract.get_edit_url()})
```

Reply:
C'est fait.